variables.tf

```diff
@@ -2,8 +2,150 @@
 ########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "resource_group_id" {
+  description = "The resource group ID where the watsonx services will be provisioned. Required when creating a new instance."
+  type        = string
+  default     = null
+  validation {
+    condition     = var.existing_watson_studio_instance_crn == null ? length(var.resource_group_id) > 0 : true
+    error_message = "You must specify a value for 'resource_group_id', required to create watson studio instance."
+  }
+
+  validation {
+    condition     = var.existing_machine_learning_instance_crn == null ? length(var.resource_group_id) > 0 : true
+    error_message = "You must specify a value for 'resource_group_id', required to create watson machine learning instance."
+  }
+}
+
+variable "prefix" {
+  description = "The name to be used on all Watson resources as a prefix."
+  type        = string
+
+  validation {
+    condition     = var.prefix != "" && length(var.prefix) <= 25
+    error_message = "You must provide a value for the resource_prefix variable and the resource_prefix length can't exceed 25 characters."
+  }
+}
+
+variable "region" {
+  default     = "us-south"
+  description = "Region where the watsonx resources will be provisioned."
+  type        = string
+
+  validation {
+    condition     = contains(["eu-de", "us-south", "eu-gb", "jp-tok", "au-syd"], var.region)
+    error_message = "You must specify `eu-de`, `eu-gb`, `jp-tok` or `us-south` as the IBM Cloud region."
+  }
+}
+
+variable "resource_tags" {
+  description = "Optional list of tags to describe the service instances created by the module."
+  type        = list(string)
+  default     = []
+}
+
+# Watson Studio
+variable "existing_watson_studio_instance_crn" {
+  default     = null
+  description = "The CRN of an existing Watson Studio instance. If not provided, a new instance will be provisioned."
+  type        = string
+}
+
+variable "watson_studio_plan" {
+  default     = "professional-v1"
+  description = "The plan that is used to provision the Watson Studio instance. The plan you choose for Watson Studio affects the features and capabilities that you can use."
+  type        = string
+  validation {
+    condition     = contains(["free-v1", "professional-v1"], var.watson_studio_plan)
+    error_message = "You must use a free-v1 or professional-v1 plan. Learn more."
+  }
+}
+
+# Watson Machine Learning
+variable "existing_machine_learning_instance_crn" {
+  default     = null
+  description = "The CRN of an existing Watson Machine Learning instance. If not provided, a new instance will be provisioned."
+  type        = string
+}
+
+variable "watson_machine_learning_plan" {
+  description = "The plan that is used to provision the Watson Machine Learning instance."
```

Comment:
* I would suggest to remove `Machine learning` and use `watsonx.ai Runtime` as this is the new name. Also, allowed values should be provided as part of the description.
* In case of `lite` plan, the option to provide service endpoints is not shown, so we can mention in the description that value is ignored and default service configuration is applied for `lite` plan.

Reply:
Name change I would like to address in a separate issue in Q1. It will disturb code at many places.